### PR TITLE
chore(compiler): move JSDoc to function definitions in transpile.ts

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This moves the JSDocs for the `transpile` and `transpileSync` functions from `public.ts` to the file where the function definitions actually are. Those functions are then re-exported from the `public.ts` file to make sure anything importing from that file will continue to work correctly.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [x] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

The JSDoc comments describing how these functions work aren't collocated with the function declaration.

## What is the new behavior?

Basically just moving some code from one place to another, there shouldn't really be any change in how things work.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/compiler/transpile.ts [hunk 1]
```diff
@@ -1,14 +1,40 @@
 import { rollupPluginUtils } from '@compiler-deps';
+import {
+  Config,
+  TransformCssToEsmInput,
+  TransformOptions,
+  TranspileOptions,
+  TranspileResults,
+} from '@stencil/core/internal';
```
Comment (tanner-reits) [inline src/compiler/transpile.ts:8]: Is changing this import what is causing the the need to update `scripts/bundles/compiler.ts`?

Comment (alicewriteswrongs) [reply src/compiler/transpile.ts:8]: not exactly, the change is necessary because now the generated `d.ts` file for `src/compiler/index.ts` depends on a type exported from `src/compiler/transpile.ts` (so in terms of build output `build/compiler/index.d.ts` depends on `build/compiler/transpile.d.ts`)

Comment (alicewriteswrongs) [reply src/compiler/transpile.ts:8]: the underlying issue is that typescript doesn't provide any functionality for bundling together `.d.ts` files and I haven't found a good solution for doing that automatically yet, so until and if we find a good way to do that when we add a type export from a module like this we'll have to make a change in the build scripts to copy the `.d.ts` over as well.

File: src/compiler/transpile.ts [hunk 2]
```diff
@@ -1,14 +1,40 @@
 import { rollupPluginUtils } from '@compiler-deps';
+import {
```
Comment (rwaskiewicz) [inline src/compiler/transpile.ts:2]: Would this also work?
```suggestion
import type {
```

**Review by rwaskiewicz — COMMENTED** [review]
LGTM, one small comment/question?

File: src/compiler/transpile.ts (hunk 2 above)
Comment (alicewriteswrongs) [reply src/compiler/transpile.ts:2]: I think so!

Comment (alicewriteswrongs) [reply src/compiler/transpile.ts]: made the change, but I accidentally squashed it into the most recent commit on the branch

**Review by rwaskiewicz — APPROVED** [review]
All good 👍
